fem/gslib.hpp

```diff
@@ -96,11 +101,24 @@ class FindPointsGSLIB
    void Interpolate(Array<unsigned int> &codes, Array<unsigned int> &proc_ids,
                     Array<unsigned int> &elem_ids, Vector &ref_pos,
                     const GridFunction &field_in, Vector &field_out);
+   void Interpolate(const GridFunction &field_in, Vector &field_out);
+   /** Search positions and interpolate */
+   void Interpolate(const Vector &point_pos, const GridFunction &field_in,
+                    Vector &field_out);
+   /** Setup FindPoints, search positions and interpolate */
+   void Interpolate(Mesh &m, const Vector &point_pos,
+                    const GridFunction &field_in, Vector &field_out);
 
    /** Cleans up memory allocated internally by gslib.
        Note that in parallel, this must be called before MPI_Finalize(), as
        it calls MPI_Comm_free() for internal gslib communicators. */
    void FreeData();
+
+   const Array<unsigned int> &GetCode()   { return gsl_code; }
+   const Array<unsigned int> &GetElem()   { return gsl_elem; }
+   const Array<unsigned int> &GetProc()   { return gsl_proc; }
+   const Vector &GetReferencePosition()   { return gsl_ref;  }
+   const Vector &GetDist2()               { return gsl_dist; }
```

Comment:
Since only constant references are being returned, these methods should be marked `const`.